scripts/merge_truth_per_tract.py

```diff
@@ -34,11 +35,11 @@ def merge_truth_per_tract(input_dir, truth_types=("truth_", "star_", "sn_"), val
     truth_types : tuple of str, optional (default: ("truth_", "star_", "sn_"))
         The types of truth objects. The string should be the prefix of truth files.
     validate : bool, optional (default: False)
-        If true, check the tract column has only one value
+        If true, check that the tract column has only one value
     silent : bool, optional (default: False)
         If true, turn off most printout.
     """
-    my_print = (lambda x: None) if silent else print
+    my_print = (lambda *x: None) if silent else print
 
     my_print("Merging all files in", input_dir)
     files_to_merge = sorted(os.listdir(input_dir))
```

Comment:
the assert on type_code should be inside the loop, so as to trigger a failure at the time the first case occurs, rather than wait for everything to be read concatenated and merged for nothing.